### DISCUSSION TITLE
Dev setup with nhost instance in local network

### DISCUSSION THREAD

**Discussion by stefan-girlich** [nhost/nhost#3041] [Q&A] [answered]
I have an nhost instance running on my computer via Docker/CLI and I want to access it with a Flutter/mobile app in development mode. Both computer and phone are on the same network.
I'd usually use [ngrok CLI](https://ngrok.com/) to expose services running on my computer on localhost to the mobile app. However, nhost creates four (auth, storage, functions, GraphQL) relevant services whose URLs I'd need to pass to the [ServiceUrls constructor](https://pub.dev/packages/nhost_dart#sample) in my app:

```
- GraphQL:          https://local.graphql.local.nhost.run
- Auth:             https://local.auth.local.nhost.run
- Storage:          https://local.storage.local.nhost.run
- Functions:        https://local.functions.local.nhost.run
```

Would I need to use ngrok on these four URLs separately or is there a more convenient way?

**Comment by stefan-girlich** [comment]
I found the relevant page in the docs: https://docs.nhost.io/guides/cli/subdomain

I passed my computer's IP as dash-separated octets (`nhost --local-subdomain 192-168-2-55  up`), but the hostname can't be resolved on the same machine:
```
> host 192-168-2-55.dashboard.local.nhost.run
Host 192-168-2-55.dashboard.local.nhost.run not found: 2(SERVFAIL)
```

Any help is appreciated

**Reply by dbarrosop** [reply, marked as answer]
The DNS resolution should work:

```
$ host 192-168-2-55.dashboard.local.nhost.run
192-168-2-55.dashboard.local.nhost.run has address 192.168.2.55
```

However, some ISPs block DNS responses that return private IP space. You can try setting a different DNS server from the one provided by your service provider (i.e. 1.1.1.1 or 8.8.8.8 or something else).

**Reply by stefan-girlich** [reply]
Setting an explicit DNS (1.1.1.1) on my workstation and mobile device worked - thanks for the quick answer! Feels like this could be worth adding to https://docs.nhost.io/guides/cli/subdomain

**Reply by dbarrosop** [reply]
Thanks for the suggestion, I just added a note there: https://github.com/nhost/nhost/pull/3046

**Comment by stefan-girlich** [comment]
@dbarrosop I'm currently seeing `net::ERR_CERT_DATE_INVALID` for the hosts `local.hasura.local.nhost.run` and `192-168-2-55.hasura.local.nhost.run`. This happens when on different devices with DNS set to `1.1.1.1`. Would you know more? Let me know if I should create a separate issue or discussion.

**Reply by dbarrosop** [reply]
my guess is you are running an old cli and the embedded certificate has expired, try upgrading to the latest CLI.

**Reply by stefan-girlich** [reply]
Updating the CLI worked, thanks
